packages/lexicon/src/validators/primitives.ts

```diff
@@ -248,39 +246,76 @@ export function string(
     }
   }
 
+  if (typeof def.format === 'string') {
+    switch (def.format) {
+      case 'datetime':
+        return datetimeFormat(path, value)
+      case 'at-uri':
+        return atUriFormat(path, value)
+      case 'did':
+        return didFormat(path, value)
+      case 'cid':
+        return cidFormat(path, value)
+    }
+  }
+
   return { success: true, value }
 }
 
-export function datetime(
-  lexicons: Lexicons,
-  path: string,
-  def: LexUserType,
-  value: unknown,
-): ValidationResult {
-  def = def as LexDatetime
+export function datetimeFormat(path: string, value: string): ValidationResult {
+  try {
+    if (!isValidISODateString(value)) {
+      throw new Error()
+    }
+  } catch {
+    return {
+      success: false,
+      error: new ValidationError(
+        `${path} must be an iso8601 formatted datetime`,
+      ),
+    }
+  }
+  return { success: true, value }
+}
 
-  // type
-  const type = typeof value
-  if (type !== 'string') {
+export function atUriFormat(path: string, value: string): ValidationResult {
+  try {
+    const uri = new AtUri(value)
+    if (!value.startsWith('at://') || uri.toString() !== value) {
+      throw new Error()
+    }
+  } catch {
     return {
       success: false,
-      error: new ValidationError(`${path} must be a string`),
+      error: new ValidationError(`${path} must be an at-uri`),
     }
   }
+  return { success: true, value }
+}
 
-  // valid iso-8601
-  {
-    try {
-      if (typeof value !== 'string' || !isValidISODateString(value)) {
-        throw new ValidationError(
-          `${path} must be an iso8601 formatted datetime`,
-        )
-      }
-    } catch {
-      throw new ValidationError(`${path} must be an iso8601 formatted datetime`)
+export function didFormat(path: string, value: string): ValidationResult {
+  const parts = value.split(':')
+  if (parts.length < 3 || parts[0] !== 'did') {
```

Comment:
We could add a check here for a missing method to avoid non-dids like `did::`.
```suggestion
  if (parts.length < 3 || parts[0] !== 'did' || !parts[1]) {
```

Reply:
yup yup 👍